[#19] feat : 기존 코드 수정

daakludens opened the pull request:
1. 유저와 게임 관련 로직 보완 및 수정
2. 게임 리스트와 특정 게임을 로그인한 모든 유저가 열람할 수 있도록 AOP 추가

daakludens committed:
[#19] feat : 로직 부분 수정, 게임 리스트를 로그인한 모든 유저가 사용할 수 있도록 AOP 추가

f-lab-dev commented on src/main/java/com/ludensdomain/aop/SessionCheck.java:
```diff
@@ -0,0 +1,11 @@
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface SessionCheck {
```
로그인 여부를 체크하는데 **로그인 방식** 은 세션뿐 아니라 다른 방식이 될 수도 있습니다. 세션이라는 이름이 들어가면 너무 종속적이 되지 않을까요?

daakludens replied:
기존에 유저 채크를 하던 LoginCheck를 로그인 확인용 aop로 이름을 변경하고, 유저 체크용은 UserCheck로 변경하겠습니다.

daakludens committed:
[#19] refactor : 기능이 명시 되도록 AOP 이름 재정의

daakludens committed:
[#19] refactor : 사용자 권한 레벨에 타입을 추가.
- db와 서버 간 데이터 매핑을 String 자료형으로 더 수월하게 함
- Enum으로 지정된 값 외에는 받지 않으면서 db에 String 자료형으로 넣어 관리되도록 변경

f-lab-dev commented on src/main/java/com/ludensdomain/aop/LoginCheckAspect.java:
```diff
@@ -18,19 +18,15 @@
-        AuthLevel role = (AuthLevel) httpSession.getAttribute(ROLE);
-        if (authLevel != role) {
+        String id = (String) httpSession.getAttribute(ID);
+        if (id.isEmpty()) {
```
`id`가 null일수도 있지 않을까요?

daakludens replied:
네, null check와 empty check 둘 다 되도록 수정했습니다.

daakludens committed:
[#19] refactor : LoginCheckAspect에 id의 null check 로직 추가

daakludens committed:
[#19] refactor : 유저 정보를 가져올 때 존재하지 않는 id를 넣은 경우를 대비해 로직 수정

f-lab-dev commented on src/main/java/com/ludensdomain/aop/RoleCheckAspect.java:
```diff
@@ -33,6 +33,6 @@ public void roleCheck(RoleCheck roleCheck) {
             throw new UnauthorizedUserException();
         }
-        log.info("{}은 해당 기능에 접근 가능한 권한입니다.", roleCheck.authLevel().getRoleName());
+        log.info("{}는/은 해당 기능에 접근 가능합니다.", roleCheck.authLevel().getRoleName());
```
info로 하면 트래픽이 많이 들어왔을 때 과도한 로그가 출력될 가능성이 생기지 않을까요?

daakludens replied:
info 로그 레벨에 유저가 접근할 때마다 찍히면 어플리케이션에 부담이 됩니다. 로그를 따로 관리하는 서버나 DB를 두는 방법도 있지만, 특별한 요구 사항이 아닌 이상 해당 로그 레벨을 사용하는 건 의미가 없어보입니다.
따라서 접근 권한이 없는 유저의 시도에 예외가 발생할 때만 로그가 찍히도록 하고 레벨은 error로 주었습니다.

f-lab-dev commented on src/main/java/com/ludensdomain/service/UserService.java:
```diff
@@ -19,23 +19,20 @@
-            return null;
+        if (user != null) {
+            String decryptPw = stringEncryptor.decrypt(user.getPassword());
+            return (password.equals(decryptPw)) ? user : null;
```
```suggestion
            return password.equals(decryptPw) ? user : null;
```

daakludens replied:
불필요한 괄호가 있었네요. 수정하겠습니다.

daakludens committed:
[#19] RoleCheckAspect에 log가 출력 수정, 삼항연산식 불필요한 괄호 제거

f-lab-dev commented on src/main/java/com/ludensdomain/aop/RoleCheckAspect.java:
```diff
@@ -31,8 +31,8 @@ public void roleCheck(RoleCheck roleCheck) {
         String role = roleCheck.authLevel().getRole();
         String sessionRole = (String) httpSession.getAttribute(ROLE);
         if (!(role.equals(sessionRole))) {
+            log.error("{}은/는 해당 기능에 접근 불가능합니다.", roleCheck.authLevel().getRoleName());
```
접근권한 없이 찔러보는식의 요청은 많이 올 것 같습니다~ 크리티컬하진 않으니 로그레벨을 낮추는것도 괜찮아보입니다~

daakludens replied:
application 흐름이나 설정 확인이 아니면서(info) 심각한 에러(error)가 아니기에 그 중간 로그 레벨인 warn으로 변경해봤습니다.

daakludens committed:
[#19] log level 변경

daakludens committed:
[#19] refactor : deleteSchedule 메서드 수정

daakludens committed:
Revert "[#19] refactor : deleteSchedule 메서드 수정"

This reverts commit 9f60ae6a

daakludens committed:
[#19] refactor : deleteSchedule() 수정
